exonum-java-binding/core/rust/src/storage/db.rs

```diff
@@ -141,95 +157,118 @@ mod tests {
-        check_ref_fork(&mut view);
-        check_owned_ref(&mut view);
+        let snapshot = db.snapshot();
+        let view = View::from_ref_snapshot(&*snapshot);
+        check_snapshot(view.get());
+        assert!(!view.can_convert_into_fork());
     }
 
     #[test]
-    fn create_view_with_owned_snapshot() {
+    fn snapshot_owned_view() {
         let db = setup_database();
         let snapshot = db.snapshot();
-        let mut view = View::from_owned_snapshot(snapshot);
-        check_ref_snapshot(&mut view);
-        check_owned_ref(&mut view);
+        let view = View::from_owned_snapshot(snapshot);
+        check_snapshot(view.get());
+        assert!(!view.can_convert_into_fork());
     }
 
     #[test]
-    fn create_view_with_ref_fork() {
+    fn fork_ref_view() {
         let db = setup_database();
         let fork = db.fork();
-        let mut view = View::from_ref_fork(&fork);
-        check_ref_fork(&mut view);
-        assert!(view.owned.is_none());
+        let view = View::from_ref_fork(&fork);
+        check_fork(view.get());
+        assert!(!view.can_convert_into_fork());
     }
 
     #[test]
-    fn create_view_with_ref_snapshot() {
+    fn fork_mut_ref_view() {
         let db = setup_database();
-        let snapshot = db.snapshot();
-        let mut view = View::from_ref_snapshot(&*snapshot);
-        check_ref_snapshot(&mut view);
-        assert!(view.owned.is_none());
+        let mut fork = db.fork();
+        let view = View::from_ref_mut_fork(&mut fork);
+        check_fork(view.get());
+        assert!(!view.can_convert_into_fork());
     }
 
-    // Creates database with a prepared state.
-    fn setup_database() -> TemporaryDB {
-        let db = TemporaryDB::new();
+    #[test]
+    fn fork_owned_view() {
+        let db = setup_database();
         let fork = db.fork();
-        entry(&fork).set(TEST_VALUE);
-        db.merge(fork.into_patch()).unwrap();
-        db
+        let view = View::from_owned_fork(fork);
+        check_fork(view.get());
+        assert!(view.can_convert_into_fork());
     }
 
-    fn entry<T>(view: T) -> Entry<T, i32>
-    where
-        T: IndexAccess,
-    {
-        Entry::new("test", view)
+    #[test]
+    fn convert_fork_into_patch() {
+        let db = TemporaryDB::new();
+        let fork = db.fork();
+        let view = View::from_owned_fork(fork);
+        let _patch = view.into_fork().into_patch();
     }
 
-    fn check_ref_fork(view: &mut View) {
-        match *view.get() {
-            ViewRef::Fork(fork) => check_value(fork),
-            _ => panic!("View::reference expected to be Fork"),
+    #[test]
+    fn get_mut_fork() {
+        let db = TemporaryDB::new();
+        let mut fork = db.fork();
+        let view = View::from_ref_mut_fork(&mut fork);
+        let mock_method = |_: &mut Fork| {};
+        match view {
+            View::RefMutFork(fork_ref) => {
+                mock_method(fork_ref);
+            }
+            _ => panic!(),
```

Comment:
What is tested here and what the panics mean (also in other methods)?

Reply:
It's just a pattern matching that has only one valid variant (and other variants must lead to test failure). I'd add panic message explaining the purpose.